x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsPhysicalOperationProviders.java

```diff
@@ -75,16 +76,17 @@ public final PhysicalOperation fieldExtractPhysicalOperation(FieldExtractExec fi
 
             DataType dataType = attr.dataType();
             String fieldName = attr.name();
-            List<BlockDocValuesReader.Factory> factories = BlockReaderFactories.factories(
-                searchContexts,
-                fieldName,
-                EsqlDataTypes.isUnsupported(dataType)
-            );
+            List<BlockLoader> loaders = BlockReaderFactories.loaders(searchContexts, fieldName, EsqlDataTypes.isUnsupported(dataType));
+            List<IndexReader> readers = searchContexts.stream().map(s -> s.searcher().getIndexReader()).toList();
 
             int docChannel = previousLayout.get(sourceAttr.id()).channel();
 
             op = op.with(
-                new ValuesSourceReaderOperator.ValuesSourceReaderOperatorFactory(factories, docChannel, fieldName),
+                new ValuesSourceReaderOperator.Factory(
+                    List.of(new ValuesSourceReaderOperator.FieldInfo(fieldName, loaders)),
```

Comment:
NOTE: ESQL won't see a performance boost out of this until we modify the planning to group sequences of field loads together and then make this a multi-element list.

Reply:
👍 should be a straight forward change

Reply:
The changes here are not complex indeed, I have a local branch with the fixes needed already.
Let's merge this one first and then I'll submit a follow-up PR